File: android/src/main/java/com/getcapacitor/plugin/spotlightsearch/SpotlightSearchPlugin.java
package com.getcapacitor.plugin.spotlightsearch;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SpotlightSearch")
public class SpotlightSearchPlugin extends Plugin {

    @PluginMethod
    public void indexItems(PluginCall call) {
        call.reject("Not implemented on Android.");
    }

    @PluginMethod
    public void indexItem(PluginCall call) {
        call.reject("Not implemented on Android.");
    }

    @PluginMethod
    public void getIndexedItemsIdentifiers(PluginCall call) {
        call.reject("Not implemented on Android.");
    }

    @PluginMethod
    public void deleteItemsWithId(PluginCall call) {
        call.reject("Not implemented on Android.");
    }

    @PluginMethod
    public void deleteAllItems(PluginCall call) {
        call.reject("Not implemented on Android.");
    }
}

File: ios/Plugin/SpotlightSearchPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(SpotlightSearchPlugin, "SpotlightSearch",
           CAP_PLUGIN_METHOD(indexItems, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(indexItem, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getIndexedItemsIdentifiers, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(deleteItemsWithId, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(deleteAllItems, CAPPluginReturnPromise);
)
